src/ext.rs

```diff
@@ -0,0 +1,79 @@
+use core::{
+  future::Future,
+  pin::Pin,
+  task::{Context, Poll},
+};
+
+use crate::AsyncDatagram;
+
+/// An extension trait that adds utility methods to [`AsyncDatagram`] types.
+pub trait AsyncDatagramExt: AsyncDatagram {
+  /// Creates a future that will send a given message to the specified target.
+  ///
+  /// The returned future will resolve to the number of bytes actually sent once the operation
+  /// completes.
+  fn send_to<'a>(
+    self: &'a mut Self,
```

Comment:
```suggestion
    &'a mut self,
```

Reply:
Whoops, forgot I could simplify this after removing a `Pin`.